lms/djangoapps/instructor/views/api.py

```diff
@@ -726,6 +728,343 @@ def extract_user_info(user):
+        'success': success
+    }
+    return JsonResponse(response_payload)
+
+
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('instructor')
+def get_temp_queries(_request, course_id):
+    """
+    Returns the temporary user queries per course
+    """
+    course_id = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+    saved_temp = data_access.get_temp_queries(course_id)
+    cleaned_queries = []
+    for query in saved_temp:
+        cleaned_queries.append({'id': query.id,
+                                'inclusion': REVERSE_INCLUSION_MAP[query.inclusion],
+                                'block_type': query.module_state_key.block_type,
+                                'block_id': query.module_state_key.block_id,
+                                'filter_on': query.filter_on,
+                                'display_name': query.entity_name,
+                                'type': query.type,
+                                'done': query.done,
+                                })
+    response_payload = {
+        'course_id': course_id.to_deprecated_string(),
+        'queries': cleaned_queries
+    }
+    return JsonResponse(response_payload)
+
+
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('instructor')
+def get_saved_queries(_request, course_id):
+    """
+    Returns all the user-saved queries per course
+    """
+    course_id = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+    groups, queries, relations = data_access.get_saved_queries(course_id)
+    cleaned_queries = []
+    relation_map = {}
+    created = {}
+    for relation in relations:
+        relation_map[relation.query_id] = relation.grouped_id
+    for group in groups:
+        created[group.id] = group.created.strftime("%m-%d-%y %H:%M")
+    for query in queries:
+        if query.id in relation_map:
+            group_id = relation_map[query.id]
+            cleaned_queries.append({'inclusion': REVERSE_INCLUSION_MAP[query.inclusion],
+                                    'block_type': query.module_state_key.block_type,
+                                    'block_id': query.module_state_key.block_id,
+                                    'filter_on': query.filter_on,
+                                    'display_name': query.entity_name,
+                                    'type': query.type,
+                                    'group': group_id,
+                                    'created': created[group_id]
+                                    })
+    response_payload = {
+        'course_id': course_id.to_deprecated_string(),
+        'queries': cleaned_queries
+    }
+    return JsonResponse(response_payload)
+
+
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('instructor')
+def get_total_students(request, course_id, make_csv=False):
```

Comment:
i was a bit confused here, I thought this was get the total _number_ of students rather than all students.  Maybe `get_all_students` might be slightly better?